go-controller/pkg/factory/factory.go

```diff
@@ -1659,6 +1659,8 @@ type waitForCacheSyncer interface {
 }
 
 func waitForCacheSyncWithTimeout(factory waitForCacheSyncer, stopCh <-chan struct{}) map[reflect.Type]bool {
+	// Give some small for sync. It helps significantly reduce unit tests time
+	time.Sleep(5 * time.Millisecond)
```

Comment:
this sleep works for unit tests, but do you have some data from the real cluster? could you run a kind cluster with some extra logging around cache sync to check how it is affected?

Reply:
This function is only called on start of the factory. So effect on real production cluster should not be anyhow affected by this change. If I'm wrong then we have another level of issue on startup.

Let me know if you have any specific concerns related to this change.

Reply:
I agree with @poroh here. Worst case it will add 5ms for every informer we are going to start when hybrid-overlay is enabled.

if this small change can cut down our unit tests run time, then that is a good win IMHO. Especially when we start this factory in 100s of tests.

Reply:
let's at least document in the PR description that it may add 5 ms delay to each informer startup depending on the speed of its sync.

Reply:
@poroh can you please capture that in your commit description? thanks.